add an option for OBJ Loder to load .obj file with same part name

zomb-676 opened the pull request:
This PR adds the ability for the OBJ Loder to load .obj file with same part name  

# Why need this?

Some modeling software such as BlockBench , which allows homonym cube names .  When its model is exported as an .obj model file . The name subsequents to  .obj format command/keyworld `o` will duplicate

Due to the current OBJ Loader implementation , the collection of the parts is a `Map<String, ModelGroup>` , so the later will override the former. As a reuslt , only the last part will remain

# Implementation

add an option bool for OBJModel.Setting called `allowHomonymObjectName`  which is false by default in order to keep forward compatibility

when the option is true ,  we will check weather the same part name occours before , if it is , we re-name it by adding a positive growth after the origin name until no repeat occour
The similar way as another modeling software called Blender , it can correctly load .obj file with homonym object name

re-name only works on the `o` .obj command/keyword now

# Maybe a problem

```java
    public Optional<? extends IModelGeometryPart> getPart(String name)
    {
        return Optional.ofNullable(parts.get(name));
    }
```

after re-name , the getter method may fail to get the correct named part 
but if your model has homonym part name , you can't expect to get it by this way
you should name the needed part yourself or you could call this method  
 `public Collection<? extends IModelGeometryPart> getParts()`
and filter according to its keys by methods such as startWith to group them

zomb-676 committed:
add allowHomonymObjectName option for OBJModel.ModelSettings
the option is false by default in order to keep compatibility
rename only works on the `o` .obj command/keyword now

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=8624) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=8624) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MinecraftForge/MinecraftForge?pullRequest=8624) it.</sub>

zomb-676 committed:
Merge branch 'MinecraftForge:1.18.x' into 1.18.x

zomb-676 committed:
change re-name strategy ,
when allowHomonymObjectName is true and homonym happens
just plus char '_' and a unique number id ,
so you can just get all homonym parts by call filter and use methods like startWith
with the homonym as parameter
test done , see https://pastebin.com/GRV65QjF

sciwhiz12 commented:
Hello! @zomb-676, please leave a comment whether you wish to continue this PR and update it to the latest active version of 1.19.2 (or leave an explanation why this PR's changes are not applicable to 1.19.2). If we do not hear back from you after some time, this PR will be closed for inactivity.

If anyone else wishes to continue the work of this PR, please create a PR with a link to this PR for reference.
Thank you for your contributions. 👋
